SurfaceDevCenterManager/DevCenterApi/DevCenterHandler.cs

```diff
@@ -101,25 +106,46 @@ private string GetDevCenterBaseUrl()
                         }
                     }
 
-                    break;
-                }
+                    string content = await infoResult.Content.ReadAsStringAsync();
+                    if (infoResult.IsSuccessStatusCode)
+                    {
+                        processContent?.Invoke(content);
+                        return null;
+                    }
 
-                string content = await infoResult.Content.ReadAsStringAsync();
-                if (infoResult.IsSuccessStatusCode)
-                {
-                    processContent?.Invoke(content);
-                    return null;
-                }
+                    try
+                    {
+                        reterr = JsonConvert.DeserializeObject<DevCenterErrorReturn>(content);
+                    }
+                    catch (JsonReaderException)
+                    {
+                        // Error is in bad format, return raw
+                        reterr = new DevCenterErrorReturn()
```

Comment:
When an 'error page' is returned, has the infoResult.statusCode also been set to indicate that? 
If so, we can check the statusCode first before attempting the json parse to avoid the exception.
The exception handler is still a good idea nonetheless, for other unexpected pages returned.